backend/eventloop/shared_resource_loop/sharedresourceloop_repositorycredential.go

```diff
@@ -0,0 +1,653 @@
+			Type:    managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionValidRepositoryCredential,
+			Reason:  errorOccuredCondition.Reason,
+			Status:  metav1.ConditionFalse,
+			Message: errorOccuredCondition.Message,
+		}
+	} else {
+		err := validateRepositoryCredentials(repositoryCredential.Spec.Repository, secret)
+		if err != nil {
+			if strings.Contains(err.Error(), "not found") {
+				// Repository does not exist
+				validRepoUrlCondition = &metav1.Condition{
+					Type:    managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionValidRepositoryUrl,
+					Reason:  managedgitopsv1alpha1.RepositoryCredentialReasonInValidRepositoryUrl,
+					Status:  metav1.ConditionFalse,
+					Message: fmt.Sprintf("Repository %s does not exist: %s", repositoryCredential.Spec.Repository, err.Error()),
+				}
+				validRepoCredCondition = &metav1.Condition{
+					Type:    managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionValidRepositoryCredential,
+					Reason:  managedgitopsv1alpha1.RepositoryCredentialReasonInValidRepositoryUrl,
+					Status:  metav1.ConditionFalse,
+					Message: fmt.Sprintf("Repository %s does not exist: %s", repositoryCredential.Spec.Repository, err.Error()),
+				}
+				errorOccuredCondition = &metav1.Condition{
+					Type:    managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionErrorOccurred,
+					Reason:  managedgitopsv1alpha1.RepositoryCredentialReasonInValidRepositoryUrl,
+					Status:  metav1.ConditionTrue,
+					Message: fmt.Sprintf("Repository %s does not exist: %s", repositoryCredential.Spec.Repository, err.Error()),
+				}
+			} else {
+				// Credentials were invalid
+				validRepoUrlCondition = &metav1.Condition{
+					Type:    managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionValidRepositoryUrl,
+					Reason:  managedgitopsv1alpha1.RepositoryCredentialReasonValidRepositoryUrl,
+					Status:  metav1.ConditionTrue,
+					Message: fmt.Sprintf("Repository %s exists", repositoryCredential.Spec.Repository),
+				}
+				validRepoCredCondition = &metav1.Condition{
+					Type:    managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionValidRepositoryCredential,
+					Reason:  managedgitopsv1alpha1.RepositoryCredentialReasonInvalidCredentials,
+					Status:  metav1.ConditionFalse,
+					Message: fmt.Sprintf("Repository Credentials provided %s for Repository %s are invalid", secret.Name, repositoryCredential.Spec.Repository),
+				}
+				errorOccuredCondition = &metav1.Condition{
+					Type:    managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionErrorOccurred,
+					Reason:  managedgitopsv1alpha1.RepositoryCredentialReasonInvalidCredentials,
+					Status:  metav1.ConditionTrue,
+					Message: fmt.Sprintf("Repository Credentials provided %s for Repository %s are invalid", secret.Name, repositoryCredential.Spec.Repository),
+				}
+			}
+		} else {
+			// No errors occured
+			errorOccuredCondition = &metav1.Condition{
```

Comment:
This part of the function appears not to be covered by unit tests (correct me if I am wrong!)

Reply:
This part is not covered as it needs the Git commands to succeed with real secrets